hadoop-tools/hadoop-azure/src/site/markdown/testing_azure.md

```diff
@@ -357,6 +357,34 @@ The Huge File tests validate Azure storages's ability to handle large files —t
 Tests at this scale are slow: they are best executed from hosts running in
 the cloud infrastructure where the storage endpoint is based.
 
+##No test no review: Run different combinations of tests using the runtests.sh script
+
+This is the expected way in which the tests have to be ran before raising a PR.
```

Comment:
Change to:
To simplify the testing across various authentication and features combinations that are mandatory for a PR, script `dev-support/testrun-scripts/runTest.sh` should be used. Once the script is updated with relevant config settings for various test combinations, it will:
1. Auto-generate configs specific to each test combinations
2. Run tests for all combinations
3. Summarize results across all the test combination runs.

As a pre-requiste step, fill config values for test accounts and credentials needed for authentication in `src/test/resources/azure-auth-keys.xml.template`  and rename as `src/test/resources/azure-auth-keys.xml`.

**To add a new test combination**: Mandatory test combinations for PR validation are already pre-filled in `dev-support/testrun-scripts/runTest.sh`. If a new one needs to be added, add a combination set within `dev-support/testrun-scripts/runTest.sh` similar to the ones already defined and
1. Provide a new combination name
2. Update properties and values array which need to be effective for the test combination
3. Call generateConfigs

**To run PR validation**: Running command 
- `runTest.sh -testCombination #combinationname#` or `runTest.sh -tc #combinationname#` : will generate configurations effective for that test combination and also run the test.
- `runTest.sh -testCombination all` or `runTest.sh -tc all` : will generate configurations for each of the combinations defined and run tests.

**Test logs**: <Put in the para starting "Once the tests are completed", with the last line modified as :
When run for PR validation, the consolidated test results needs to be pasted into the PR comment section.>

**To generate config for use in IDE**: Running command 
`runTest.sh -generateConfig #combinationname#` or `runTest.sh -gc #combinationname#` 
will update the effective config relevant for the specific test combination. Hence the same config files used by the mvn test runs can be used for IDE without any manual updates needed within config file.

**Other command line options**: 
1. Thread count : ABFS mvn tests are run in parallel mode. Tests by default are run with 8 thread count. It can be changed by providing -t #ThreadCount#

Reply:
Actually rewrite the Test logs section as below:
Test runs will create a folder within `dev-support/testlogs` to save the test logs. Folder name will be the test start timestamp. The mvn test command line logs for each combination will be saved into a file as `Test-Logs
-$combination` into this folder. Incase of any failures, this file will have the failure exception stack. 
At the end of the test run, the consolidated results of all the combination runs will be saved into a file as `Test-Results.log` in the same folder. When run for PR validation, the consolidated test results needs to be pasted into the PR comment section.

Reply:
Done